nep-12.mediawiki

```diff
@@ -0,0 +1,2130 @@
+}
+</pre>
+
+
+'''Error return value'''
+
+<pre>
+interface Error {
+  type: string;
+  description: string;
+  data: string;
+}
+</pre>
+
+
+'''Example'''
+
+<pre>
+getApplicationLog({
+  txid: '7e049fd7c253dabf38e4156df30c78b30d49f307196aa89b99a47d2330789bf2',
+  network: 'TestNet'
+})
+.then((result: Object) => {
+  console.log('Application log of transaction execution: ' + JSON.stringigy(result));
+})
+.catch(({type: string, description: string, data: any}) => {
+  switch(type) {
+    case NO_PROVIDER:
+      console.log('No provider available.');
+      break;
+   case RPC_ERROR:
+    console.log('There was an error when broadcasting this transaction to the network.');
+    break;
+  }
+});
+</pre>
+
+
+'''Provider Request Handling'''
+
+Upon receiving this request
+* Broadcast the RPC request
+* Return the RPC response results
+
+
+Example
+<pre>
+{
+  "txid": "0x7e049fd7c253dabf38e4156df30c78b30d49f307196aa89b99a47d2330789bf2",
+  "executions": [
+    {
+      "trigger": "Application",
+      "contract": "0x72985e7f2cea98b89af54d8607bc6400814c4b45",
+      "vmstate": "HALT",
+      "gas_consumed": "5.292",
+      "stack": [],
+      "notifications": [
+        {
+          "contract": "0x849d095d07950b9e56d0c895ec48ec5100cfdff1",
+          "state": {
+            "type": "Array",
+            "value": [
+              {
+                "type": "ByteArray",
+                "value": "7472616e73666572"
+              },
+              {
+                "type": "ByteArray",
+                "value": "296ac124021a71c449a9bad320c16429b08ad6ee"
+              },
+              {
+                "type": "ByteArray",
+                "value": "7869ef9732cdf6f6d54adaa5cae3b55a9396bceb"
+              },
+              {
+                "type": "ByteArray",
+                "value": "00e1f505"
+              }
+            ]
+          }
+        },
+        {
+          "contract": "0x849d095d07950b9e56d0c895ec48ec5100cfdff1",
+          "state": {
+            "type": "Array",
+            "value": [
+              {
+                "type": "ByteArray",
+                "value": "7472616e73666572"
+              },
+              {
+                "type": "ByteArray",
+                "value": "296ac124021a71c449a9bad320c16429b08ad6ee"
+              },
+              {
+                "type": "ByteArray",
+                "value": "b1fdddf658ce5ff9f83e66ede2f333ecfcc0463e"
+              },
+              {
+                "type": "ByteArray",
+                "value": "00e1f505"
+              }
+            ]
+          }
+        }
+      ]
+    }
+  ]
+}
+</pre>
+
+
+===Write Methods===
+====send====
+Invoke a transfer of a specified amount of a given asset from the connected account to another account.
+
+
+'''Method Interface'''
+
+<pre>
+function send({
+  fromAddress: string,
+  toAddress: string,
+  asset: string,
+  amount: string,
+  remark?: string,
+  fee?: string,
+  network?: string,
+  broadcastOverride?: boolean,
```

Comment:
What is `broadcastOverride`?

Reply:
I just got it... it makes the Provider not to transmit (only generate) the transaction, correct?
Perhaps, a simpler name would be better, like: `broadcast = false`.

Reply:
This is correct.

Reply:
It is called `broadcastOverride`, as the default behaviors is that the wallet will broadcast the transaction on behalf of the user directly after signing. This default behavior can be overridden to not broadcast, and just return the signed transaction.

Reply:
Ok,now I get that.